umad/src/main/java/com/pkukielka/AccessMonitorRewriter.java

```diff
@@ -144,7 +141,23 @@ private boolean isSafeMethod(ConstPool pool, int index) {
         return safeMethods.contains(fqn);
     }
 
-    private boolean isSafe(CtMethod editableMethod, String dottedName) {
+    private CtMethod getMethodIfExist(ConstPool constPool, int methodIndex)  {
+        try {
+            String className = constPool.getMethodrefClassName(methodIndex);
+            String methodName = constPool.getMethodrefName(methodIndex);
+            if (methodName.endsWith("$")) methodName = methodName.substring(0, methodName.length() - 1);
+
+            ClassPool classpool = ClassPool.getDefault();
+            CtClass editableClass = classpool.get(className);
+            CtMethod[] declaredMethods = editableClass.getDeclaredMethods(methodName);
+
+            return declaredMethods.length == 1 ? declaredMethods[0] : null;
+        } catch (NotFoundException e) {
+            return null;
+        }
+    }
+
+    private boolean isSafe(CtMethod editableMethod, boolean checkOneLevelDeeper) {
```

Comment:
Maybe we can have configurable level of deep checks? Since it + caching may work really well (this is basically my idea with unlimited safety checks)